sunpy/net/jsoc/jsoc.py

```diff
@@ -489,165 +497,131 @@ def _process_time(self, time):
         return time.datetime
 
     def _make_recordset(self, start_time, end_time, series, wavelength='',
-                        segment='', **kwargs):
-        # Build the dataset string
-        # Extract and format Wavelength
-        if wavelength:
-            if not series.startswith('aia'):
-                raise TypeError("This series does not support the wavelength attribute.")
-            else:
-                if isinstance(wavelength, list):
-                    wavelength = [int(np.ceil(wave.to(u.AA).value)) for wave in wavelength]
-                    wavelength = str(wavelength)
-                else:
-                    wavelength = '[{0}]'.format(int(np.ceil(wavelength.to(u.AA).value)))
+                        segment='', primekeys={}, **kwargs):
+        """
+        Take the query arguments and build a record string.
+        """
 
         # Extract and format segment
-        if segment != '':
+        if segment:
+            if isinstance(segment, list):
+                segment = str(segment)[1:-1].replace(' ', '').replace("'", '')
             segment = '{{{segment}}}'.format(segment=segment)
 
+        # Extract and format sample
         sample = kwargs.get('sample', '')
         if sample:
             sample = '@{}s'.format(sample)
 
-        dataset = '{series}[{start}-{end}{sample}]{wavelength}{segment}'.format(
-                   series=series, start=start_time.strftime("%Y.%m.%d_%H:%M:%S_TAI"),
-                   end=end_time.strftime("%Y.%m.%d_%H:%M:%S_TAI"),
-                   sample=sample,
-                   wavelength=wavelength, segment=segment)
-
-        return dataset
+        # Extract and format primekeys
+        pkstr = ''
+        c = drms.Client()
+        pkeys = c.pkeys(series)
 
-    def _make_query_payload(self, start_time, end_time, series, notify=None,
-                            protocol='FITS', compression='rice', **kwargs):
-        """
-        Build the POST payload for the query parameters
-        """
+        for pkey in pkeys:
+            if pkey in ['T_OBS', 'T_REC', 'T_START']:
+                pkstr += '[{start}-{end}{sample}]'.format(
+                    start=start_time.strftime("%Y.%m.%d_%H:%M:%S_TAI"),
```

Comment:
Would it not be better if the times coming in here were `astropy.time.Time` objects? rather than converting a `datetime` object which is probably mostly UTC into a string representation of TAI time?